src/driver.jl

```diff
@@ -309,6 +313,39 @@ const __llvm_initialized = Ref(false)
     return ir, (; entry, compiled)
 end
 
+# Protect null globals from being killed and inlined
+function externalize_julia_globals!(mod::LLVM.Module)
+    changed = false
+    for gbl in LLVM.globals(mod)
+        if LLVM.linkage(gbl) == LLVM.API.LLVMInternalLinkage &&
+           typeof(LLVM.initializer(gbl)) <: LLVM.PointerNull &&
+           (startswith(LLVM.name(gbl), "jl_global") ||
+            startswith(LLVM.name(gbl), "jl_sym"))
```

Comment:
Why also `jl_sym`, not just `jl_global` (in a function that externalizes globals)?